src/data/buffer/parser/basic.lean

```diff
@@ -2139,4 +2140,216 @@ end
 
 end many
 
+section nat
+
+lemma nat_of_done {val : ℕ} (h : nat cb n = done n' val) :
+  val = (nat.of_digits 10 ((((cb.to_list.drop n).take (n' - n)).reverse.map
+          (λ c, c.to_nat - '0'.to_nat)))) :=
```

Comment:
The RHS of this lemma is pretty messy - could you add a docstring to summarize the statement?

Also, why `(cb.to_list.drop n).take (n' - n)` instead of ``(cb.to_list.take n').drop n`?

Reply:
`cb.to_list.drop n` is, in my opinion, a good canonical way to express what `char_buffer` the parser is aware of when fed a particular `cb : char_buffer` and position to parse from `n`. Then, since `parser.nat` is `prog`, we know that `n ≤ n'`, so `(n' - n)` will simplify nicely when `n` and `n'` are actually known.